### ISSUE TITLE
Missing documentation for FilePath

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6046]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#6046|
|Type|task|
|Created|2012-10-04 13:05:33Z|
|Branch|https://github.com/twisted/twisted/tree/filepath-docs-6046|

The [documentation for FilePath](http://twistedmatrix.com/documents/current/api/twisted.python.filepath.FilePath.html) has some "No summary" and "Undocumented" methods listed, the attached patch fills in those blanks.

Attachments:

* [filepath-doc-6046.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ae8/ae89853ccbfe58660bc27a89574a2161c70f1f2c/3adc720880765afc9119ba1a5df050830571c7ee.patch) (21953 bytes) - added by thijs on 2012-10-04 13:05:59Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6046 6046
type__task task
reporter__thijs thijs
priority__normal normal
milestone__ 
branch__branches_filepath_docs_6046 branches/filepath-docs-6046
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1349355933000000 1349355933000000
changetime__1356382182000000 1356382182000000
version__None None
owner__thijs thijs
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_1"></a>|@tomprince set owner to @thijstriemstra|
|-|-|

* ```_PathHelper}} doesn't actually exist (it is referenced in one the doc strings

* There are quite a few places that change {{{L{```} to ```C{```} for no apparent reason. Or introduce ```C{```} where ```L{```} would work.

* [twistedchecker](https://launchpad.net/twistedchecker) reports a number of docstring errors after this patch (although less than before).

Feel free to commit after fixing.

Also,

* There are a number of places (```basename``` and ```splitext``` for example, where the description of the function and the return value are essentially the same. It would be nice to simplify one of them (probably the return value).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [36624]) Branching to 'filepath-docs-6046'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [36625]) Apply filepath-doc-6046.patch 

refs [#6046](https://github.com/twisted/twisted/issues/6046)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [36626]) adjust a bunch of C{} to L{}; fix some other minor issues

refs [#6046](https://github.com/twisted/twisted/issues/6046)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

(In [36628]) some more doc fixes, mostly pointed out by twistedchecker

refs [#6046](https://github.com/twisted/twisted/issues/6046)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/filepath-docs-6046)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [36631]) Merge filepath-docs-6046

Author: thijs, exarkun
Reviewer: tom.prince
Fixes: [#6046](https://github.com/twisted/twisted/issues/6046)

Various improvements to the API documentation content and markup of `twisted.python.filepath`.
